### PR TITLE
add datetime to min/max-value rules

### LINKED ISSUE #97: Update min/max-value rules with support for date types

### PR DESCRIPTION
**Pull request by zargot**
fixes #97

- refactor error reporting to unify coercion and normal error logic in new module 'reports'
- fix datetime parsing/conversion
- extend min/max-rule tests with datetime

### CODE REVIEW

File: src/odm_validation/reports.py [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+
+def gen_rule_error(ctx: ErrorCtx,
+                   err_template: Optional[str] = None,
+                   kind: Optional[ErrorKind] = None,
+                   ) -> dict:
+    """
+    :param err_template: overrides rule error-template
+    :param kind: overrides ctx.kind
+    """
+    kind = kind or ctx.kind
```
Comment (yulric) [inline src/odm_validation/reports.py:104]: ErrorCtx does not have a `kind` field?

**Review by yulric — CHANGES_REQUESTED** [review]
1. error-report-4.json should be reverted since its being used in the documentation and its not consistent with the dataset the report is for.
2. Ok, so one issue with the changes you made to all the assets is that the documentation is using them which means it will be confusing to users.

File: src/odm_validation/reports.py (hunk 1 above)
Comment (zargot) [reply src/odm_validation/reports.py:104]: done

**Review by yulric — CHANGES_REQUESTED** [review]
1. Can you add tests for generating version 1 schemas?

**Review by yulric — APPROVED** [review]

**Pull request merged**